ogr/ogrsf_frmts/miramon/ogrmiramondatasource.cpp

```diff
@@ -0,0 +1,310 @@
+        case wkbMultiSurfaceM:
+        case wkbCurveM:
+        case wkbSurfaceM:
+        case wkbPolyhedralSurfaceM:
+        case wkbTINM:
+        case wkbTriangleM:
+        case wkbPointZM:
+        case wkbLineStringZM:
+        case wkbPolygonZM:
+        case wkbMultiPointZM:
+        case wkbMultiLineStringZM:
+        case wkbMultiPolygonZM:
+        case wkbGeometryCollectionZM:
+        case wkbCircularStringZM:
+        case wkbCompoundCurveZM:
+        case wkbCurvePolygonZM:
+        case wkbMultiCurveZM:
+        case wkbMultiSurfaceZM:
+        case wkbCurveZM:
+        case wkbSurfaceZM:
+        case wkbPolyhedralSurfaceZM:
+        case wkbTINZM:
+        case wkbTriangleZM:
+            CPLError(CE_Warning, CPLE_NotSupported,
+                     "Measures in this layer will be ignored.");
+            break;
+        default:
+            break;
+    }
+
+    /* -------------------------------------------------------------------- */
+    /*    If the dataset has an extension, it is understood that the path   */
+    /*       of the file is where to write, and the layer name is the       */
+    /*       dataset name (without extension).                              */
+    /* -------------------------------------------------------------------- */
+    const char *pszExtension = CPLGetExtension(pszRootName);
+    char *pszFullMMLayerName;
+    if (EQUAL(pszExtension, "pol") || EQUAL(pszExtension, "arc") ||
+        EQUAL(pszExtension, "pnt"))
+    {
+        char *pszMMLayerName;
+        pszMMLayerName = CPLStrdup(CPLResetExtension(pszRootName, ""));
+        pszMMLayerName[strlen(pszMMLayerName) - 1] = '\0';
+
+        pszFullMMLayerName = CPLStrdup((const char *)pszMMLayerName);
+
+        // Checking that the folder where to write exists
+        const char *szDestFolder = CPLGetDirname(pszFullMMLayerName);
+        char **papszDirContent = nullptr;
+        papszDirContent = VSIReadDir(szDestFolder);
+        if (!papszDirContent)
+        {
+            CPLFree(pszMMLayerName);
+            CPLFree(pszFullMMLayerName);
+            CPLError(CE_Failure, CPLE_AppDefined,
+                     "The folder %s does not exist.", szDestFolder);
+            return nullptr;
+        }
+        else
+            CSLDestroy(papszDirContent);
+        CPLFree(pszMMLayerName);
+    }
+    else
+    {
+        const char *osPath;
+
+        osPath = pszRootName;
+        pszFullMMLayerName =
+            CPLStrdup(CPLFormFilename(pszRootName, pszLayerName, ""));
+
+        /* -------------------------------------------------------------------- */
+        /*      Let's create the folder if it's not already created.            */
+        /*      (only the las level of the folder)                              */
+        /* -------------------------------------------------------------------- */
+        char **papszDirContent = nullptr;
+        papszDirContent = VSIReadDir(osPath);
+        if (!papszDirContent)
```

Comment:
```suggestion
        VSIStatBufL sStat;
        if (VSIStatL(osPath.c_str(), &sStat) != 0 || !VSI_ISDIR(sStat.st_mode))
```

Reply:
if (VSIStatL(osPath, &sStat) instead of if (VSIStatL(osPath.c_str, &sStat) cause it's const char * and not a std:str